src/header.rs

```diff
@@ -1,9 +1,13 @@
-use serde::{Deserialize, Serialize};
+use std::result;
+
+use serde::{de, Deserialize, Serialize};
 
 use crate::algorithms::Algorithm;
 use crate::errors::Result;
 use crate::serialization::b64_decode;
 
+type X509Chain = Vec<Vec<u8>>;
```

Comment:
Doesn't this type need to be public? I think I would rather skip it tbh as it's only used in this file and the actual type is clearer

Reply:
The public-with-private type checker automatically replaces the type with private type aliases in these types of cases when the aliased type is public. I removed it.